File: app/src/main/java/com/stan/dagger_android/news/ActivityNews.java
package com.stan.dagger_android.news;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.TextView;

import com.stan.dagger_android.R;
import com.stan.dagger_android.base.BaseActivity;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.OnClick;

/**
 * Created by uu on 2017/12/29.
 */

public class ActivityNews extends BaseActivity implements ActivityNewsContract.View {

    public static final String TAG = "ActivityNews";

    @Inject
    ActivityNewsContract.Presenter presenter;

    @BindView(R.id.message)
    TextView message;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_news);
    }

    @OnClick(R.id.message)
    public void clickMessage() {
        presenter.getMessage("getMessage!");
    }

    @Override
    protected void onResume() {
        super.onResume();
        presenter.takeView(this);
    }

    @Override
    protected void onPause() {
        super.onPause();
        presenter.dropView();
    }

    @Override
    public void showMessage(String message) {
        this.message.setText(message);
    }

}

File: app/src/main/java/com/stan/dagger_android/news/ActivityNews.kt
package com.stan.dagger_android.news

import android.os.Bundle
import android.widget.TextView
import butterknife.BindView
import butterknife.OnClick
import com.stan.dagger_android.R
import com.stan.dagger_android.base.BaseActivity
import javax.inject.Inject

/**
 * Created by uu on 2017/12/29.
 */

class ActivityNews : BaseActivity(), ActivityNewsContract.View {

    @Inject
    lateinit var presenter: ActivityNewsContract.Presenter

    @BindView(R.id.message)
    lateinit var message: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_news)
    }

    @OnClick(R.id.message)
    fun clickMessage() {
        presenter.getMessage("getMessage!")
    }

    override fun onResume() {
        super.onResume()
        presenter.takeView(this)
    }

    override fun onPause() {
        super.onPause()
        presenter.dropView()
    }

    override fun showMessage(message: String) {
        this.message.text = message
    }

    companion object {
        val TAG = "ActivityNews"
    }

}
